### ISSUE TITLE
HTTPS and SignalR hubs?

### ISSUE THREAD

**Issue by kmanev073** [aspnet/SignalR#2481]
I am using SignalR 1.0.0 with .NET Core 2.1 and it seems I can't map the hub to https address.
When I try to connect to https://mysiteurl/myhub I get negotiation failed and 404 error.
When I try to connect to the same url but with http only it works.
 I used the documentation from msdn and I looked through your examples but I could not find a solution. Hope someone can help. Thanks in advacne! That's my code on the server:

```
services.AddSignalR();
//and
app.UseSignalR(routes =>
{
    routes.MapHub<MainHub>("/mainhub");
});
```
Basically, everything as explained on msdn. and my JS code:
```
const connection = new signalR.HubConnectionBuilder()
        .withUrl("/mainhub")
        .build();

    connection.start().catch(err => console.error(err.toString()));
```

And the error in the browser:
[https://i.imgur.com/27TeXmP.png](url)

Reagards,
Kaloyan Manev

**Comment by davidfowl** [comment]
myhub or mainhub?

**Comment by ferronsw** [comment]
I've no problem connection over https. Is your application forced to use https?

Mine does like discribed in this [article ](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-2.1&tabs=visual-studio)

**Comment by kmanev073** [comment]
I did what @ferronsw 's article said and it works now. I was not using app.UseHttpsRedirection(); and instead I had my own redirection implemented. It seems it is needed thanks. :)
